### ISSUE TITLE
ls-hubd crashes after patching permissions

### ISSUE THREAD

**Issue by Informatic** [webosbrew/webos-homebrew-channel#33]
@rhinoswirl noticed some weird behaviour after manually applying changes in #24 on a device running webOS `4.3.0-9703 / 20210107010305 / HE_DTV_W18A_AFADABAA`

* Likely `ls-control scan-services` run in `start-devmode.sh` on every boot was causing a bus crash
* Changing `sleep 2` back to `sleep 3` sometimes helped (?)
* After commenting out `elevate-service` call a factory reset screen popped up, but wasn't able to progress through the language/country selection screen (likely ls-hubd crash), telnetd was still up
* After nuncommenting the `elevate-service` call and a reboot a factory reset screen was on again, but it was possible to progress further, telnetd was still up.
* `ares-install` was causing a crash as well, if permissions were set to `all`

This needs some further investingation and testing on other webOS versions.

**Comment by Informatic** [comment]
We have a confirmation that this *didn't* happen after applying #24 manually on webOS 5.2.1-2521.

**Comment by rhinoswirl** [comment]
I have some additional details on the reason why that factory reset happened:

https://github.com/rhinoswirl/webostv-research/blob/main/self-troubleshoot/factory-reset.md

This still doesn't explain why luna crashed.

@Ruthenic can you install updates using ares when the service is running with the `all` permissions set?

I see that there are some changes in the script and the replace doesn't run in every boot but just once. I don't know if this can be related.

**Comment by Ruthenic** [comment]
@rhinoswirl I can install applications using `ares-install` whilst the service is running with `all` set, assuming this is what you mean by updates

**Comment by rhinoswirl** [comment]
I think that my issue can be explained because of a set of unfortunate circumstances.

After reading my old messages, I see that I said that it didn't happen all the time before, but started happening always after changing the sleep time (the script on #24 was different back then).

Maybe it could be that I was doing `ares-install`s for updating the service while the `start-devmode.sh` script was doing its thing with `scan-services` and sending an ares-install to a service while it was still being initialized was what was actually causing the crash of `ls-hubd`. 

With the script now checking if there are any differences before running `scan-services` or, in this case, `scan-volatile-dirs`, it should no longer cause any issues incase a noob like me does the same steps I did in the timings that I did them.

So, I'd say this issue can be closed. It's very unlikely that it will ever happen again to someone else.

As of now, I can't use the service properly but this is likely related with the factory reset that happened.
